java/src/Ice/src/main/java/com/zeroc/Ice/ObjectPrx.java

```diff
@@ -665,7 +665,7 @@ static String ice_staticId() {
    * @return <code>obj</code>.
    */
   static ObjectPrx checkedCast(ObjectPrx obj) {
-    return obj;
+    return checkedCast(obj, noExplicitContext);
```

Comment:
For `checkedCast` on `ObjectPrx`s, we used to just return the object as-is.
This was an optimization since we know ever proxy inherits from `ObjectPrx`.

@bernardnormier says that `checkedCast` should _always_ trigger a call to `ice_isA`.
So, I removed this optimization, now `checkedCast` will always trigger a remote call.
(Unless the proxy you're casting is `null`, then there's nothing we can do...)